### DISCUSSION TITLE
Vercel + Cloudflare Proxy = Incorrect location recognition

### DISCUSSION THREAD

**Discussion by andriiPalko** [vercel/community#2532] [Help]
Hello, I tried to follow this [example](https://github.com/vercel/examples/tree/main/edge-middleware/geolocation) to get the user's geolocation headers but experienced the issue when my site is behind Cloudflare Proxy I am getting geolocation of Cloudflare Proxy IP address but not of a real user. As a workaround, I tried to get geolocation data from [Cloudflare headers](https://developers.cloudflare.com/fundamentals/get-started/reference/http-request-headers/#cf-ipcountry) but figured out that Vercel is not transferring Cloudflare headers to my NextJS application.

Is there anything I can do to make this work?

**Comment by amyegan** [comment]
Hi @andriiPalko. It looks like you're already on the right track to solve this. I believe the solution is to add a location header (don't forget to [enable the Add visitor location headers Managed Transform](https://developers.cloudflare.com/rules/transform/managed-transforms/configure/) if using `CF-IPCountry`) and read the header in your Middleware to direct visitor appropriately.

**Reply by andriiPalko** [reply]
Hi @amyegan, thank you for your response. I tried what you are suggesting, but it seems like Vercel router doesn't pass Cloudflare headers to my application (and correspondingly to middleware). Are you sure it is passing those headers? If so confirm it please and I will search for the issue on my side.
